### PR TITLE
GEM cluster data vs emulator plots in L1T DQM [11_3_X]

### PR DESCRIPTION
**Pull request by dildick**
#### PR description:

With this PR we add data vs emulator comparison histograms for GEM trigger clusters. These are 1D histograms for each cluster property and "data-emulator" plots - which should be empty. 

At this point we do not yet have GEM clusters in the Run-3 RAW data. However, we anticipate GEM clusters to be there very soon through either the CSC unpacker, or the EMTF unpacker. At that point, I'll update the "data" inputTag to `muonCSCDigis` or `emtfStage2Digis`.

#### PR validation:

Code largely borrowed from #33085. Tested with WF 11634.0

```
11634.0_TTbar_14TeV+2021+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST+ALCA Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED  - time date Sat Mar 27 21:15:27 2021-date Sat Mar 27 20:43:58 2021; exit: 0 0 0 0 0
1 1 1 1 1 tests passed, 0 0 0 0 0 failed
```

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

No backport needed yet.

FYI @jshlee @akhter-towsifa

### CODE REVIEW

**Comment by jfernan2** [comment]
please test

**Comment by dildick** [comment]
Looks like a similar issue as in #33120. I need to separate Run-2 from Run-3 for the Stage-2 L1T sequence in the DQM.

**Comment by dildick** [comment]
This fixes WF 1330.0 
```
1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15_L1TMuDQM+HARVESTUP15_L1TMuDQM+NANOUP15 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED  - time date Sun Mar 28 22:42:38 2021-date Sun Mar 28 22:31:32 2021; exit: 0 0 0 0 0
1 1 1 1 1 tests passed, 0 0 0 0 0 failed
```

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
please test

**Comment by dildick** [comment]
Output from ```runTheMatrix.py -l limited```

```
4.22_RunCosmics2011A+RunCosmics2011A+RECOCOSD+ALCACOSD+SKIMCOSD+HARVESTDC Step0-DAS_ERROR Step1-NOTRUN Step2-NOTRUN Step3-NOTRUN Step4-NOTRUN  - time date Mon Mar 29 17:13:37 2021-date Mon Mar 29 17:12:52 2021; exit: 1 0 0 0 0
4.53_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Mon Mar 29 17:48:55 2021-date Mon Mar 29 17:12:53 2021; exit: 0 0 0 0
5.1_TTbar+TTbarFS+HARVESTFS Step0-PASSED Step1-PASSED  - time date Mon Mar 29 17:34:29 2021-date Mon Mar 29 17:12:53 2021; exit: 0 0
7.3_CosmicsSPLoose_UP18+CosmicsSPLoose_UP18+DIGICOS_UP18+RECOCOS_UP18+ALCACOS_UP18+HARVESTCOS_UP18 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED  - time date Mon Mar 29 17:32:37 2021-date Mon Mar 29 17:12:54 2021; exit: 0 0 0 0 0
8.0_BeamHalo+BeamHalo+DIGICOS+RECOCOS+ALCABH+HARVESTCOS Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED  - time date Mon Mar 29 17:40:01 2021-date Mon Mar 29 17:13:38 2021; exit: 0 0 0 0 0
9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Mon Mar 29 17:48:59 2021-date Mon Mar 29 17:32:38 2021; exit: 0 0 0 0
25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED  - time date Mon Mar 29 17:59:30 2021-date Mon Mar 29 17:34:29 2021; exit: 0 0 0 0 0
135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS Step0-PASSED Step1-PASSED Step2-PASSED  - time date Mon Mar 29 17:49:18 2021-date Mon Mar 29 17:40:01 2021; exit: 0 0 0
136.731_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Mon Mar 29 18:24:24 2021-date Mon Mar 29 17:48:59 2021; exit: 0 0 0 0
136.7611_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM Step0-PASSED Step1-PASSED Step2-PASSED  - time date Mon Mar 29 17:59:19 2021-date Mon Mar 29 17:49:00 2021; exit: 0 0 0
136.793_RunDoubleEG2017C+RunDoubleEG2017C+HLTDR2_2017+RECODR2_2017reHLT_skimDoubleEG_Prompt+HARVEST2017 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Mon Mar 29 18:36:17 2021-date Mon Mar 29 17:49:19 2021; exit: 0 0 0 0
136.8311_RunJetHT2017F_reminiaod+RunJetHT2017F_reminiaod+REMINIAOD_data2017+HARVEST2017_REMINIAOD_data2017 Step0-PASSED Step1-PASSED Step2-PASSED  - time date Mon Mar 29 18:16:41 2021-date Mon Mar 29 17:59:19 2021; exit: 0 0 0
136.8523_RunJetHT2018C_nanoULremini+RunJetHT2018C_nanoULremini+NANOEDM2018_106Xv2+HARVESTNANOAOD2018_106Xv2 Step0-PASSED Step1-PASSED Step2-PASSED  - time date Mon Mar 29 18:18:21 2021-date Mon Mar 29 17:59:30 2021; exit: 0 0 0
136.874_RunEGamma2018C+RunEGamma2018C+HLTDR2_2018+RECODR2_2018reHLT_skimEGamma_Offline_L1TEgDQM+HARVEST2018_L1TEgDQM Step0-PASSED Step1-FAILED Step2-NOTRUN Step3-NOTRUN  - time date Mon Mar 29 18:21:56 2021-date Mon Mar 29 18:16:41 2021; exit: 0 21504 0 0
136.88811_RunJetHT2018D_reminiaodUL+RunJetHT2018D_reminiaodUL+REMINIAOD_data2018UL+HARVEST2018_REMINIAOD_data2018UL Step0-PASSED Step1-FAILED Step2-NOTRUN  - time date Mon Mar 29 18:25:35 2021-date Mon Mar 29 18:18:21 2021; exit: 0 21504 0
140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI Step0-PASSED Step1-PASSED Step2-PASSED  - time date Mon Mar 29 18:40:20 2021-date Mon Mar 29 18:21:57 2021; exit: 0 0 0
140.56_RunHI2018+RunHI2018+RECOHID18+HARVESTDHI18 Step0-PASSED Step1-FAILED Step2-NOTRUN  - time date Mon Mar 29 18:32:11 2021-date Mon Mar 29 18:24:25 2021; exit: 0 21504 0
158.01_HydjetQ_reminiaodPbPb2018_INPUT+HydjetQ_reminiaodPbPb2018_INPUT+REMINIAODHI2018PPRECOMB+HARVESTHI2018PPRECOMINIAOD Step0-DAS_ERROR Step1-NOTRUN Step2-NOTRUN  - time date Mon Mar 29 18:25:37 2021-date Mon Mar 29 18:25:35 2021; exit: 1 0 0
1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Mon Mar 29 18:48:05 2021-date Mon Mar 29 18:25:37 2021; exit: 0 0 0 0
1325.81_TTbar_13_106Xv1NanoAODINPUT+TTbar_13_106Xv1NanoAODINPUT+NANOEDMMC2017_106XMiniAODv1+HARVESTNANOAODMC2017_106XMiniAODv1 Step0-DAS_ERROR Step1-NOTRUN Step2-NOTRUN  - time date Mon Mar 29 18:32:13 2021-date Mon Mar 29 18:32:12 2021; exit: 1 0 0
1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15_L1TMuDQM+HARVESTUP15_L1TMuDQM+NANOUP15 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED  - time date Mon Mar 29 18:56:16 2021-date Mon Mar 29 18:32:13 2021; exit: 0 0 0 0 0
101.0_SingleElectronE120EHCAL+SingleElectronE120EHCAL Step0-PASSED  - time date Mon Mar 29 18:38:46 2021-date Mon Mar 29 18:36:17 2021; exit: 0
312.0_Pyquen_ZeemumuJets_pt10_2760GeV_2021+Pyquen_ZeemumuJets_pt10_2760GeV_2021+DIGIHI2021MIX+RECOHI2021MIX+HARVESTHI2021PPRECO Step0-PASSED Step1-PASSED Step2-FAILED Step3-NOTRUN  - time date Mon Mar 29 19:11:12 2021-date Mon Mar 29 18:38:47 2021; exit: 0 0 35584 0
25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25+NANOUP15_PU25 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED  - time date Mon Mar 29 19:21:51 2021-date Mon Mar 29 18:40:21 2021; exit: 0 0 0 0 0
1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED  - time date Mon Mar 29 19:19:26 2021-date Mon Mar 29 18:48:05 2021; exit: 0 0 0 0 0
1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVDSIPIXELCALRUN1+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED Step5-PASSED Step6-PASSED Step7-PASSED Step8-PASSED  - time date Mon Mar 29 19:23:01 2021-date Mon Mar 29 18:56:17 2021; exit: 0 0 0 0 0 0 0 0 0
10024.0_TTbar_13+2017+TTbar_13TeV_TuneCUETP8M1_GenSim+Digi+RecoFakeHLT+HARVESTFakeHLT+ALCA+Nano Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED Step5-PASSED  - time date Mon Mar 29 19:45:28 2021-date Mon Mar 29 19:11:12 2021; exit: 0 0 0 0 0 0
10042.0_ZMM_13+2017+ZMM_13TeV_TuneCUETP8M1_GenSim+Digi+RecoFakeHLT+HARVESTFakeHLT+ALCA+Nano Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED Step5-PASSED  - time date Mon Mar 29 19:44:17 2021-date Mon Mar 29 19:19:27 2021; exit: 0 0 0 0 0 0
10224.0_TTbar_13+2017PU+TTbar_13TeV_TuneCUETP8M1_GenSim+DigiPU+RecoFakeHLTPU+HARVESTFakeHLTPU+Nano Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED  - time date Mon Mar 29 19:58:49 2021-date Mon Mar 29 19:21:51 2021; exit: 0 0 0 0 0
10824.0_TTbar_13+2018+TTbar_13TeV_TuneCUETP8M1_GenSim+Digi+RecoFakeHLT+HARVESTFakeHLT+ALCA+Nano Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED Step5-PASSED  - time date Mon Mar 29 19:54:22 2021-date Mon Mar 29 19:23:02 2021; exit: 0 0 0 0 0 0
11634.0_TTbar_14TeV+2021+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST+ALCA Step0-PASSED Step1-PASSED Step2-FAILED Step3-NOTRUN Step4-NOTRUN  - time date Mon Mar 29 20:18:46 2021-date Mon Mar 29 19:44:17 2021; exit: 0 0 35584 0 0
12434.0_TTbar_14TeV+2023+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST+ALCA Step0-PASSED Step1-PASSED Step2-FAILED Step3-NOTRUN Step4-NOTRUN  - time date Mon Mar 29 20:17:24 2021-date Mon Mar 29 19:45:28 2021; exit: 0 0 35584 0 0
23234.0_TTbar_14TeV+2026D49+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Mon Mar 29 20:59:32 2021-date Mon Mar 29 19:54:22 2021; exit: 0 0 0 0
28234.0_TTbar_14TeV+2026D60+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Mon Mar 29 20:55:00 2021-date Mon Mar 29 19:58:50 2021; exit: 0 0 0 0
34634.0_TTbar_14TeV+2026D76+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Mon Mar 29 21:03:36 2021-date Mon Mar 29 20:17:25 2021; exit: 0 0 0 0
34834.999_TTbar_14TeV+2026D76PU_PMXS1S2PR+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+PREMIX_PremixHLBeamSpot14PU+DigiTriggerPU+RecoGlobalPU+HARVESTGlobalPU Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED  - time date Mon Mar 29 21:20:10 2021-date Mon Mar 29 20:18:46 2021; exit: 0 0 0 0 0
250202.181_TTbar_13UP18+TTbar_13UP18+PREMIXUP18_PU25+DIGIPRMXLOCALUP18_PU25+RECOPRMXUP18_PU25+HARVESTUP18_PU25 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED  - time date Mon Mar 29 21:30:34 2021-date Mon Mar 29 20:55:01 2021; exit: 0 0 0 0 0
34 30 26 21 13 4 1 1 1 tests passed, 3 3 3 0 0 0 0 0 0 failed
```

**Comment by dildick** [comment]
Repeated 3 tests after the last commit
```
312.0_Pyquen_ZeemumuJets_pt10_2760GeV_2021+Pyquen_ZeemumuJets_pt10_2760GeV_2021+DIGIHI2021MIX+RECOHI2021MIX+HARVESTHI2021PPRECO Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Tue Mar 30 08:13:57 2021-date Tue Mar 30 07:47:52 2021; exit: 0 0 0 0
11634.0_TTbar_14TeV+2021+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST+ALCA Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED  - time date Tue Mar 30 08:21:30 2021-date Tue Mar 30 07:47:53 2021; exit: 0 0 0 0 0
12434.0_TTbar_14TeV+2023+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST+ALCA Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED  - time date Tue Mar 30 08:17:51 2021-date Tue Mar 30 07:47:53 2021; exit: 0 0 0 0 0
3 3 3 3 2 tests passed, 0 0 0 0 0 failed
```

**Comment by jfernan2** [comment]
@dildick is this needed for the next MWGR? If so, we need a backport to 11_2_X asap

**Comment by jfernan2** [comment]
+1

**Comment by dildick** [comment]
@jfernan2 I propose that we merge this PR before https://github.com/cms-sw/cmssw/pull/33085, which has more work to do.

**Comment by dildick** [comment]
@silviodonato Can you sign off, please?

File: DQM/L1TMonitor/python/L1TStage2Emulator_cff.py [hunk 1]
```diff
@@ -47,6 +47,13 @@
 valOmtfDigis.srcCSC = cms.InputTag('omtfStage2Digis')
 valOmtfDigis.srcRPC = cms.InputTag('omtfStage2Digis')
 
+# GEM TPG
+from L1Trigger.L1TGEM.simGEMDigis_cff import *
+valMuonGEMPadDigis = simMuonGEMPadDigis.clone()
+valMuonGEMPadDigis.InputCollection = cms.InputTag('muonGEMDigis')
+valMuonGEMPadDigiClusters = simMuonGEMPadDigiClusters.clone()
+valMuonGEMPadDigiClusters.InputCollection = cms.InputTag('valMuonGEMPadDigis')
```
Comment (silviodonato) [inline DQM/L1TMonitor/python/L1TStage2Emulator_cff.py:55]: It is safer to use
```
valMuonGEMPadDigis = simMuonGEMPadDigis.clone( InputCollection = 'muonGEMDigis' )
valMuonGEMPadDigiClusters = simMuonGEMPadDigiClusters.clone( InputCollection = 'valMuonGEMPadDigis')
```

**Comment by silviodonato** [comment]
+1
I've added a minor comment that you can include in #33085.

I was confused by the PR description 

> #33085 needs to be merged first.

File: DQM/L1TMonitor/python/L1TStage2Emulator_cff.py (hunk 1 above)
Comment (dildick) [reply DQM/L1TMonitor/python/L1TStage2Emulator_cff.py:55]: Ok. I'll include it.

**Pull request merged**